File: services/ai/app/api/health.py
from fastapi import APIRouter
from datetime import datetime

router = APIRouter()


@router.get("/")
async def health_check():
    """Health check endpoint"""
    return {
        "status": "healthy",
        "timestamp": datetime.utcnow().isoformat(),
        "service": "ai",
    }


@router.get("/live")
async def liveness():
    """Liveness probe"""
    return {"status": "ok"}


@router.get("/ready")
async def readiness():
    """Readiness probe - check dependencies here"""
    return {"status": "ready"}

File: apps/api/src/routes/health.ts
import { FastifyInstance, FastifyRequest, FastifyReply } from 'fastify';

export async function healthRoutes(fastify: FastifyInstance) {
  fastify.get('/health', async (_request: FastifyRequest, reply: FastifyReply) => {
    return reply.send({
      status: 'healthy',
      timestamp: new Date().toISOString(),
      service: 'api',
    });
  });

  fastify.get('/health/live', async (_request: FastifyRequest, reply: FastifyReply) => {
    return reply.send({ status: 'ok' });
  });

  fastify.get('/health/ready', async (_request: FastifyRequest, reply: FastifyReply) => {
    // Add checks for dependencies here
    return reply.send({ status: 'ready' });
  });
}
